### PR TITLE
Genesis 6582

### PR DESCRIPTION
**Pull request by seainair**

### CODE REVIEW

**Comment by Neuromancer42** [comment]
> **Verdict: Request changes** -- The kernel does not handle the q-seqlen dimension within the attention computation (no per-token masking against kv positions), which will produce incorrect results for seq_len > 1.

注意这个问题，decode N个token应当产生[N, kv_len]的causal mask，而非[1, kv_len]

File: mojo_opset/tests/accuracy/operators/test_attention.py [hunk 1]
```diff
@@ -37,15 +38,21 @@ def generate_paged_decode_data(
...
+        query = torch.randn(batch_size, num_q_heads, head_dim, dtype=dtype)
     if max_seq_len > 0:
         total_seq_lens = torch.randint(0, max_seq_len, (batch_size,), dtype=torch.int32)
-        total_seq_lens = torch.clamp(total_seq_lens, min=1)
     else:
         total_seq_lens = torch.randperm(batch_size, dtype=torch.int32)
 
+    # the seqlens of kv cache must larger than seq_lens of query
+    min_seq_len = seq_len if seq_len != -1 else 1
+    total_seq_lens = torch.clamp(total_seq_lens, min=min_seq_len)
```
Comment (Neuromancer42) [inline mojo_opset/tests/accuracy/operators/test_attention.py]: randperm的分支是为了测试padding的，建议恢复成：
```
min_seq_len = seq_len if seq_len != -1 else 1
if max_seq_len > 0:
  total_seq_lens = torch.randint(0, max_seq_len, (batch_size,), dtype=torch.int32)
  total_seq_lens = torch.clamp(total_seq_lens, min=min_seq_len)
else:
  total_seq_lens = torch.randperm(batch_size, dtype=torch.int32)
  total_seq_lens = torch.where(total_seq_lens == 0, 0, total_seq_lens + min_seq_len - 1)
```

**Review by Neuromancer42 — APPROVED** [review]
LGTM
